homeassistant/components/sesame/lock.py

```diff
@@ -77,17 +75,19 @@ def unlock(self, **kwargs) -> None:
 
     def update(self) -> None:
         """Update the internal state of the device."""
-        self._sesame.update_state()
+        status = self._sesame.get_status()
         self._nickname = self._sesame.nickname
-        self._api_enabled = self._sesame.api_enabled
-        self._is_unlocked = self._sesame.is_unlocked
-        self._device_id = self._sesame.device_id
-        self._battery = self._sesame.battery
+        self._device_id = str(self._sesame.id)
+        self._serial = self._sesame.serial
+        self._battery = status['battery']
+        self._is_locked = status['locked']
+        self._responsive = status['responsive']
 
     @property
     def device_state_attributes(self) -> dict:
         """Return the state attributes."""
         attributes = {}
         attributes[ATTR_DEVICE_ID] = self._device_id
+        attributes[ATTR_SERIAL_NO] = self._serial
```

Comment:
We don't allow serial numbers and similar static info in state attributes. The state attributes should hold info that is mostly dynamic and related to the state of the entity.